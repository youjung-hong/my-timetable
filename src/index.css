body {
  line-height: 2;
}
