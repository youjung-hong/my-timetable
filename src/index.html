<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title><%= htmlWebpackPlugin.options.title %></title>

</head>
<body style="font-family: sans-serif;">
    <div id="root"></div>
    <script>
        document.addEventListener('DOMContentLoaded', () => {
            const timetable = new MyTimetable(document.getElementById('root'), [{
                id: 1,
                color: '#51FF0D',
                startAt: '2022-01-02T07:15:00',
                endAt: '2022-01-02T07:20:00',
            }, {
                id: 2,
                color: '#FF0DCA',
                startAt: '2022-06-20T18:10:00',
                endAt: '2022-06-20T18:55:00'
            }], {
                startHour: 5
            });

            setTimeout(() => {
                timetable.addTimeline({
                id: 3,
                color: '#FFBB0D',
                startAt: '2022-06-20T10:00:00',
                endAt: '2022-06-20T10:20:00'
            });
            }, 1500);

            setTimeout(() => {
                timetable.addTimeline({
                    id: 2,
                    color: '#FF0DCA',
                    startAt: '2022-06-20T08:30:00',
                    endAt: '2022-06-20T09:00:00'
                });
            }, 3000);

            setTimeout(() => {
                timetable.removeTimeline(3);
                timetable.removeTimeline(5);
            }, 4500);
        })
    </script>
</body>
</html>