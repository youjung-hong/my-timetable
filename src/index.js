import { TaskItem } from './TaskItem/TaskItem'
import { TaskItemList } from './TaskItemList/TaskItemList'
import { Heatmap } from './Heatmap/Heatmap'
import './index.css'

export { TaskItem, TaskItemList, Heatmap }
