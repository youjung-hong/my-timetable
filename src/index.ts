import { Timetable } from './Timetable'
import './index.css'

export default Timetable
