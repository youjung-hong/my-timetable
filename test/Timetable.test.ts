import { Timetable } from '../src/Timetable';

beforeEach(() => {
    document.body.innerHTML = '<div id="root"></div>';
})

// 초기화
// - rootElement가 없으면 에러난다.
// - rootelement가 있으면, 그 아래에 생성된다.
// - timelines가 없어도 된다. 
// - timelines가 있으면 추가된다.
// - options가 없어도 된다.
// - options.startHour가 있으면, 시작 시점이 변경된다. 
test('initialization OK: new Timetable(rootElement)', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement);
    expect(timetable).toBeInstanceOf(Timetable);
    expect(timetable.getTimelines()).toEqual([]);
    expect(rootElement.outerHTML).toMatchSnapshot();
})

test('initialization Error: new Timetable(null)', () => {
    expect(() => {
        const rootElement = document.getElementById('root1') as HTMLDivElement;
        new Timetable(rootElement);
    }).toThrowError();
})

test('initialization OK: new Timetable(rootElement, timelines)', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable1 = new Timetable(rootElement, []);
    expect(timetable1).toBeInstanceOf(Timetable);
    expect(timetable1.getTimelines()).toEqual([]);
    expect(rootElement.outerHTML).toMatchSnapshot();

    const timetable2 = new Timetable(rootElement, [{
        id: 1,
        color: 'yellow',
        startAt: '2022-06-23T07:30:00',
        endAt: '2022-06-23T09:00:00',
    }]);
    expect(timetable2).toBeInstanceOf(Timetable);
    expect(timetable2.getTimelines().length).toEqual(1);
    expect(timetable2.getTimelines()[0].id).toEqual(1);
    expect(timetable2.getTimelines()[0].color).toEqual('yellow');
    expect(timetable2.getTimelines()[0].startAt).toEqual(new Date('2022-06-23T07:30:00'));
    expect(timetable2.getTimelines()[0].endAt).toEqual(new Date('2022-06-23T09:00:00'));
    expect(timetable2.getTimelines()[0].meta).toBeUndefined();
    expect(rootElement.outerHTML).toMatchSnapshot();
});

test('initialization OK: new Timetable(rootElement, timelines, options)', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement, [], {
        startHour: 5
    });
    expect(timetable).toBeInstanceOf(Timetable);
    expect(timetable.getTimelines()).toEqual([]);
    expect(rootElement.outerHTML).toMatchSnapshot();
})

// addTimeline: 추가한다 // 수정한다
// - 값이 없는 것을 추가하면 새로 생긴다.화면에도 추가된다.
// - 값이 있는 것을 추가하면 수정한다.화면에도 수정된다 
test('addTimelines: create new timelines', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement);
    timetable.addTimelines([{
        id: 3,
        color: 'blue',
        startAt: '2022-06-20T10:30:00',
        endAt: '2022-06-20T11:15:00'
    }]);

    expect(timetable.getTimelines().length).toEqual(1);
    expect(timetable.getTimelines()[0].id).toEqual(3);
    expect(timetable.getTimelines()[0].color).toEqual('blue');
    expect(timetable.getTimelines()[0].startAt).toEqual(new Date('2022-06-20T10:30:00'));
    expect(timetable.getTimelines()[0].endAt).toEqual(new Date('2022-06-20T11:15:00'));
    expect(timetable.getTimelines()[0].meta).toBeUndefined();
    expect(rootElement.outerHTML).toMatchSnapshot();
});

test('addTimelines: modify existing timelines', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement, [{
        id: 1,
        color: 'yellow',
        startAt: '2022-06-23T07:30:00',
        endAt: '2022-06-23T09:00:00',
    }]);

    timetable.addTimelines([{
        id: 1,
        color: 'blue',
        startAt: '2022-06-23T11:00:00',
        endAt: '2022-06-23T12:15:00'
    }]);

    expect(timetable.getTimelines().length).toEqual(1);
    expect(timetable.getTimelines()[0].id).toEqual(1);
    expect(timetable.getTimelines()[0].color).toEqual('blue');
    expect(timetable.getTimelines()[0].startAt).toEqual(new Date('2022-06-23T11:00:00'));
    expect(timetable.getTimelines()[0].endAt).toEqual(new Date('2022-06-23T12:15:00'));
    expect(timetable.getTimelines()[0].meta).toBeUndefined();
    expect(rootElement.outerHTML).toMatchSnapshot();
})

// removeTimeline: 지운다 
// - id가 있는 애들을 넣는다, 삭제된다. 화면에서도 지워진다. 
// - id가 안 맞는 애들을 넣으면, 아무것도 변하지 않는다. 
test('removeTimelines: remove existing timelines', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement, [{
        id: 1,
        color: 'yellow',
        startAt: '2022-06-23T07:30:00',
        endAt: '2022-06-23T09:00:00',
    }]);

    timetable.removeTimelines([1]);

    expect(timetable.getTimelines().length).toEqual(0);
    expect(rootElement.outerHTML).toMatchSnapshot();
})

test('removeTimelines: cannot remove timelines which is not exist inside timetable', () => {
    const rootElement = document.getElementById('root') as HTMLDivElement;
    const timetable = new Timetable(rootElement, [{
        id: 1,
        color: 'yellow',
        startAt: '2022-06-23T07:30:00',
        endAt: '2022-06-23T09:00:00',
    }]);

    timetable.removeTimelines([2]);

    expect(timetable.getTimelines().length).toEqual(1);
    expect(rootElement.outerHTML).toMatchSnapshot();
})
