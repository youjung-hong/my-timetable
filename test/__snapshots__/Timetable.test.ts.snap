// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addTimeline: create a new timeline 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10<div><div class=\\"timeline\\" style=\\"left: 50%; width: calc(50% - 1px); background-color: blue;\\" data-timeline-id=\\"3\\"></div></div></div><div class=\\"hour\\" data-hour=\\"11\\">11<div><div class=\\"timeline\\" style=\\"left: 0%; width: calc(25% - 1px); background-color: blue;\\" data-timeline-id=\\"3\\"></div></div></div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`addTimeline: modify a existing timeline 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11<div><div class=\\"timeline\\" style=\\"left: 0%; width: calc(100% - 1px); background-color: blue;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"12\\">12<div><div class=\\"timeline\\" style=\\"left: 0%; width: calc(25% - 1px); background-color: blue;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`initialization OK: new Timetable(rootElement) 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`initialization OK: new Timetable(rootElement, timelines) 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`initialization OK: new Timetable(rootElement, timelines) 2`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7<div><div class=\\"timeline\\" style=\\"left: 50%; width: calc(50% - 1px); background-color: yellow;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"8\\">8<div><div class=\\"timeline\\" style=\\"left: 0%; width: calc(100% - 1px); background-color: yellow;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`initialization OK: new Timetable(rootElement, timelines, options) 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div></div></div></div></div>"
`;

exports[`removeTimeline: cannot remove a existing timeline 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7<div><div class=\\"timeline\\" style=\\"left: 50%; width: calc(50% - 1px); background-color: yellow;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"8\\">8<div><div class=\\"timeline\\" style=\\"left: 0%; width: calc(100% - 1px); background-color: yellow;\\" data-timeline-id=\\"1\\"></div></div></div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;

exports[`removeTimeline: remove a existing timeline 1`] = `
"<div id=\\"root\\"><div class=\\"timetable\\"><div class=\\"hours hour-header\\"><div class=\\"hour\\">0</div><div class=\\"hour\\">1</div><div class=\\"hour\\">2</div><div class=\\"hour\\">3</div><div class=\\"hour\\">4</div><div class=\\"hour\\">5</div><div class=\\"hour\\">6</div><div class=\\"hour\\">7</div><div class=\\"hour\\">8</div><div class=\\"hour\\">9</div><div class=\\"hour\\">10</div><div class=\\"hour\\">11</div><div class=\\"hour\\">12</div><div class=\\"hour\\">13</div><div class=\\"hour\\">14</div><div class=\\"hour\\">15</div><div class=\\"hour\\">16</div><div class=\\"hour\\">17</div><div class=\\"hour\\">18</div><div class=\\"hour\\">19</div><div class=\\"hour\\">20</div><div class=\\"hour\\">21</div><div class=\\"hour\\">22</div><div class=\\"hour\\">23</div></div><div class=\\"contents\\">
            <div class=\\"minutes\\">
                <div class=\\"minute minute-0\\">00</div>
                <div class=\\"minute minute-10\\">10</div>
                <div class=\\"minute minute-20\\">20</div>
                <div class=\\"minute minute-30\\">30</div>
                <div class=\\"minute minute-40\\">40</div>
                <div class=\\"minute minute-50\\">50</div>
            </div><div class=\\"hours\\"><div class=\\"hour\\" data-hour=\\"0\\">0</div><div class=\\"hour\\" data-hour=\\"1\\">1</div><div class=\\"hour\\" data-hour=\\"2\\">2</div><div class=\\"hour\\" data-hour=\\"3\\">3</div><div class=\\"hour\\" data-hour=\\"4\\">4</div><div class=\\"hour\\" data-hour=\\"5\\">5</div><div class=\\"hour\\" data-hour=\\"6\\">6</div><div class=\\"hour\\" data-hour=\\"7\\">7</div><div class=\\"hour\\" data-hour=\\"8\\">8</div><div class=\\"hour\\" data-hour=\\"9\\">9</div><div class=\\"hour\\" data-hour=\\"10\\">10</div><div class=\\"hour\\" data-hour=\\"11\\">11</div><div class=\\"hour\\" data-hour=\\"12\\">12</div><div class=\\"hour\\" data-hour=\\"13\\">13</div><div class=\\"hour\\" data-hour=\\"14\\">14</div><div class=\\"hour\\" data-hour=\\"15\\">15</div><div class=\\"hour\\" data-hour=\\"16\\">16</div><div class=\\"hour\\" data-hour=\\"17\\">17</div><div class=\\"hour\\" data-hour=\\"18\\">18</div><div class=\\"hour\\" data-hour=\\"19\\">19</div><div class=\\"hour\\" data-hour=\\"20\\">20</div><div class=\\"hour\\" data-hour=\\"21\\">21</div><div class=\\"hour\\" data-hour=\\"22\\">22</div><div class=\\"hour\\" data-hour=\\"23\\">23</div></div></div></div></div>"
`;
